ilastik/applets/labeling/labelingGui.py

```diff
@@ -636,10 +637,19 @@ def _addNewLabel(self):
         newRow = self._labelControlUi.labelListModel.rowCount()
         self._labelControlUi.labelListModel.insertRow( newRow, label )
 
-        if self._allowDeleteLastLabelOnly:
-            # make previous label unremovable
+        if self._allowDeleteLastLabelOnly and self._defTwoInitialLabels:
+            # make previous label unremovable when we have at least three labels
             if newRow > 2:
                 self._labelControlUi.labelListModel.makeRowPermanent(newRow - 1)
+        elif self._allowDeleteLastLabelOnly:
+            # make previous label unremovable
+            if newRow > 0:
+                self._labelControlUi.labelListModel.makeRowPermanent(newRow - 1)
+        elif self._defTwoInitialLabels:
+            # if there are only two labels remaining make the unremovable
```

Comment:
the comment is a bit misleading. shouldn't it say something like "if a third label is added, allow all to be deleted"...